src/test/java/org/minions/devfund/katerinaanzoleaga/battleship/ShipTest.java

```diff
@@ -0,0 +1,280 @@
+        final int columnBattleCruiserSeven = 7;
+
+        battleCruiser.placeShipAt(1, 1, true, theOcean);
+        battleCruiser.shootAt(1, 1);
+        assertTrue(battleCruiser.getHit()[0]);
+        assertTrue(battleCruiser.shootAt(1, 2));
+        assertTrue(battleCruiser.shootAt(1, columnBattleCruiserThree));
+        assertTrue(battleCruiser.shootAt(1, columnBattleCruiserFour));
+        assertTrue(battleCruiser.shootAt(1, columnBattleCruiserFive));
+        assertTrue(battleCruiser.shootAt(1, columnBattleCruiserSix));
+        assertTrue(battleCruiser.shootAt(1, columnBattleCruiserSeven));
+        assertTrue(battleCruiser.isSunk());
+        assertFalse(battleCruiser.shootAt(1, columnBattleCruiserSeven + 1));
+        assertFalse(battleCruiser.shootAt(2, columnBattleCruiserSix));
+    }
+
+
+    /**
+     * Test several shots until sunk a destroyer.
+     */
+    @Test
+    public void testisSunkDestroyer() {
+        destroyer.placeShipAt(0, 1, true, theOcean);
+        assertEquals(true, destroyer.shootAt(0, 1));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(false, destroyer.shootAt(1, 0));
+        assertEquals(true, destroyer.shootAt(0, 2));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(false, destroyer.shootAt(0, destroyer.getLength() + 1));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(true, destroyer.shootAt(0, destroyer.getLength()));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(true, destroyer.shootAt(0, destroyer.getLength() - 1));
+        assertEquals(true, destroyer.isSunk());
+
+        theOcean.print();
+    }
+
+    /**
+     * Test several shots until sunk a submarine.
+     */
+    @Test
+    public void testSunkSubmarine() {
+        final int row = 0;
+        final int column = 0;
+        final int columnSubmarineZero = 0;
+        final int columnSubmarineOne = 1;
+        final int columnSubmarineTwo = 2;
+        final Ship ship = new Submarine();
+        ship.placeShipAt(row, column, true, theOcean);
+        ship.shootAt(row, columnSubmarineZero);
+        ship.shootAt(row, columnSubmarineOne);
+        ship.shootAt(row, columnSubmarineTwo);
+        ship.shootAt(row, columnSubmarineZero);
+        assertTrue(ship.isSunk());
+    }
+    /**
+     * Test several shots until sunk a vertical destroyer.
+     */
+    @Test
+    public void testSunkDestroyerH() {
+        final Ship destroyer = new Destroyer();
+        destroyer.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < destroyer.getLength(); ++i) {
+            destroyer.shootAt(0, i);
+            if (i != destroyer.getLength() - 1) {
+                assertFalse(destroyer.isSunk());
+            }
+        }
+        assertTrue(destroyer.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal destroyer.
+     */
+    @Test
+    public void testSunkDestroyerV() {
+        final Ship destroyer = new Destroyer();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "destroyer" which hides the field declared at line 22. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)